core/src/test/scala/unit/kafka/admin/DeleteOffsetsConsumerGroupCommandTest.scala

```diff
@@ -0,0 +1,185 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kafka.admin
+
+import java.util.Properties
+import java.util.concurrent.ExecutionException
+
+import kafka.server.Defaults
+import kafka.utils.TestUtils
+import org.apache.kafka.clients.consumer.ConsumerConfig
+import org.apache.kafka.clients.consumer.KafkaConsumer
+import org.apache.kafka.clients.producer.KafkaProducer
+import org.apache.kafka.clients.producer.ProducerConfig
+import org.apache.kafka.clients.producer.ProducerRecord
+import org.apache.kafka.common.TopicPartition
+import org.apache.kafka.common.protocol.Errors
+import org.apache.kafka.common.serialization.ByteArrayDeserializer
+import org.apache.kafka.common.serialization.ByteArraySerializer
+import org.apache.kafka.common.utils.Utils
+import org.junit.Test
+import org.junit.Assert._
+
+class DeleteOffsetsConsumerGroupCommandTest extends ConsumerGroupCommandTest {
+
+  def getArgs(group: String, topic: String): Array[String] = {
+    Array(
+      "--bootstrap-server", brokerList,
+      "--delete-offsets",
+      "--group", group,
+      "--topic", topic
+    )
+  }
+
+  @Test
+  def testDeleteOffsetsNonExistingGroup(): Unit = {
+    val group = "missing.group"
+    val topic = "foo:1"
+
+    val service = getConsumerGroupService(getArgs(group, topic))
+
+    try {
+      service.deleteOffsets(group, List(topic))
+      fail("GroupIdNotFoundException should have been raised")
+    } catch {
+      case e: ExecutionException =>
+        if (e.getCause != Errors.GROUP_ID_NOT_FOUND.exception())
+          throw e
+    }
+  }
+
+  @Test
+  def testDeleteOffsetsWithTopicPartition(): Unit = {
+    val producer = createProducer()
+    try {
+      producer.send(new ProducerRecord(topic, 0, null, null)).get()
+    } finally {
+      Utils.closeQuietly(producer, "producer")
+    }
+
+    val consumer = createConsumer()
+    try {
+      TestUtils.subscribeAndWaitForRecords(this.topic, consumer)
```

Comment:
this is known to be flaky/fragile. Do we need to produce and consume before committing? What happens if we commit offsets there were not consumed yet?

Reply:
We need to have a Stable consumer group with an active member here in order to have the `GROUP_SUBSCRIBED_TO_TOPIC` error back.

Note that I have changed the default settings to avoid rebalance during the tests so it should be fine: https://github.com/apache/kafka/pull/7362/files#diff-aaaa208bc59b8c742438185e63d79aa9R178